activesupport/lib/active_support/message_verifier.rb

```diff
@@ -198,8 +203,35 @@ def decode(data)
       end
 
       def generate_digest(data)
-        require "openssl" unless defined?(OpenSSL)
-        OpenSSL::HMAC.hexdigest(OpenSSL::Digest.const_get(@digest).new, @secret, data)
+        OpenSSL::HMAC.hexdigest(@digest, @secret, data)
+      end
+
+      def digest_length
+        # In hexadecimal (AKA base16) it takes 4 bits to represent a character,
```

Comment:
I feel like this is the kind of code that needs an explanation, but I don't know rails' policy towards inline comments. Happy to change this.